modules/core/src/main/scala/mapping.scala

```diff
@@ -17,6 +17,26 @@ trait Mapping[F[_]] {
   val schema: Schema
   val typeMappings: List[TypeMapping]
 
+  def validate: Boolean = {
+    val typeMappingNames: List[String] = typeMappings.flatMap(_.tpe.asNamed.toList.map(_.name))
+
+    def typeMappingFieldNames: List[String] = typeMappings.collect {
+      case om: ObjectMapping => om.fieldMappings.flatMap {
+        case _: CursorField[_] => Nil
+        case _: CursorAttribute[_] => Nil
+        case fm => List(fm.fieldName)
+      }
+    }.flatten
+
+    val mappingTypesExistInSchema: Boolean =
+      typeMappingNames.forall(name => schema.types.map(_.name).contains(name))
```

Comment:
It would be good to have some kind of logging or feedback about which types and fields are missing.

Reply:
Maybe have `validate` return a list of errors encoded as json? See `QueryInterpreter.mkOneError`.

Reply:
Sounds good to me.